src/ngraph/pass/cse.cpp

```diff
@@ -190,38 +213,31 @@ static unordered_map<type_index, function<bool(shared_ptr<Node>, shared_ptr<Node
 class NodeKey
 {
 public:
-    NodeKey(shared_ptr<Node> n,
+    NodeKey(const shared_ptr<Node>& n,
```

Comment:
No need for a shared pointer since the node is owned throughout the pass.
No need for `m_node_ref` and `m_ti`. Just use `n->type_info()` and you can compare them by pointer.

Reply:
It does need to be a shared_ptr since a share_ptr is passes to the cse_<*> functions defined and those functions can be defined by the user of CSE as the CPU backend does. I can't change that without changing the CPU backend and that is beyond the scope of this PR.
I'm not trying to cleanup the code, just make it faster, and it is much faster.